packages/memory/src/utils/serialize.ts

```diff
@@ -0,0 +1,24 @@
+const encoder = new TextEncoder();
+const decoder = new TextDecoder();
+
+/**
+ * Serialize a given JavaScript value to it's Uint8Array equivalent
+ *
+ * @param val - JS value to serialize
+ * @returns Uint8Array of serialized value
+ */
+export const serialize = (val: unknown) => {
+	const valueString = JSON.stringify(val);
+	return encoder.encode(valueString);
+};
+
+/**
+ * Deserialize a given Uint8Array into a JavaScript value
+ *
+ * @param val - Binary array to deserialize
+ * @returns JavaScript value
+ */
+export const deserialize = <T = unknown>(val: Uint8Array) => {
+	const valueString = decoder.decode(val);
+	return <T>JSON.parse(valueString);
```

Comment:
Not sure is there an actual need to use the `parseJSON` util here? Just asking to be on the safer side.

Reply:
Clever! I forgot we added that 🙂